lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -601,7 +601,6 @@ def write(chunk)
             end
 
           rescue GRPC::BadStatus => error
-            increment_failed_requests_count(error.code)
```

Comment:
Quoting a comment from the other PR:
```
It looks like when the retry limit is reached, the output plugin just doesn't get invoked, so the buffer data will be discarded silently. We may have to keep track of the retry counter ourselves, and on the last retry increment the "dropped" entry count instead.
```

Some obvious solutions to that issue:
1. Keep track of a `map` (key: `request`, value: `tried_count`), compare the `tried_count` towards `retry_limit`. `GoogleCloudOutput`'s parent class `BufferedOutput` has a variables `@disable_retry_limit` and `@retry_limit`.
Pros: The counter should be quite accurate.
Cons: Additional CPU/memory usage and potential latency due to the computation we need.
Open question: We need to "flush" the mapped request when they are sent successful. That probably means we will compare every successful request against the `map`. Or maybe we simple set a TTL, and flush older `request`.

2. Add a `increment_retried_requests_count`metrics separately.
Pros: Minimum additional CPU/memory usage or latency. Users should be able to rely on this counter to detect `forever retrying issues`.
Cons: The `failed_requests_count` counter will not be very accurate. Requests that exceed `retry_limit` will not be counted.

WDYT?

Reply:
Also mentioned this in the internal ticket. Maybe we can discuss it there. Will open a separate PR to address that.

Reply:
SG, let's discuss on a separate PR.